src/P2P/Node.hs

```diff
@@ -585,26 +587,43 @@ newSession conf node = do
             let env = peerClientEnv node newPeerInfo
             (info, newSes) <- mask $ \restore -> do
                 now <- getCurrentTimeIntegral
-                t <- R.randomRIO
-                    ( round (0.9 * timeoutMs)
-                    , round (1.1 * timeoutMs)
-                    )
-                !newSes <- async $ restore $ timeout t
+                t <- sessionTimeout $ secondsToTimeSpan $ _p2pConfigSessionTimeout conf
+                !newSes <- async $ restore $ timeout (int $ timeSpanToMicros t)
                     $ _p2pNodeClientSession node (loggFun node) env newPeerInfo
                 incrementActiveSessionCount peerDb newPeerInfo
                 !info <- atomically $ addSession node newPeerInfo newSes now
                 return (info, newSes)
             logg node Debug $ "Started peer session " <> showSessionId newPeerInfo newSes
             loggFun node Info $ JsonLog info
   where
-    TimeSpan timeoutMs = secondsToTimeSpan @Double (_p2pConfigSessionTimeout conf)
     peerDb = _p2pNodePeerDb node
 
     syncFromPeer_ pinfo
         | _p2pConfigPrivate conf = return True
         | _p2pNodeDoPeerSync node = syncFromPeer node pinfo
         | otherwise = return True
 
+-- | (Roughly) exponentially distributed timespans with the given expectation
+-- within the range of a a tenth of the expectation and ten times the
+-- expectation.
+--
+-- The expected value of the actual distribution gets increasinly imprecise
+-- as the input value gets smaller, because a minimum result of 5 seconds
+-- is implemented. Input values of less than 30s should be avoided.
+--
+sessionTimeout :: TimeSpan Int -> IO (TimeSpan Int)
+sessionTimeout expected = do
+    x <- exponential (1 / us) :: IO Double
```

Comment:
Use `System.Random.MWC` instead, it's better in basically every way -- including already shipping with a truncated exponential distribution. I used it for exactly this in https://github.com/kadena-io/chainweb-node/pull/1254/files, you could crib the code from there

Reply:
I'd like to use it. However, I run into compatibility issues with our rather old nix snapshot. I don't remember the details. However, this implementation should be the same than what is used in mac-random. Also, I think, modern versions of the random package have much better performance than previous versions.

Reply:
Lars -- mwc-random was already in the deps list?